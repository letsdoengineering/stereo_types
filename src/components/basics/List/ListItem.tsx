import React from 'react'
import getClassNames from '../../../utils/get-class-names'
import listItemStyles from './ListItem.module.css'

type ListItemProps = React.AllHTMLAttributes<HTMLLIElement> & {
  /** unique key for each <li> rendered to help react track dynamically rendered contents */
  listKey: string
  /** optional: styles a list to appear as links if set - defaults to false */
  isLinks?: boolean
}

/** Text: renders a Text tag of a type based on the tagName prop */
const ListItem: React.FC<ListItemProps> = ({
  listKey,
  className,
  isLinks = false,
  children,
  ...rest
}: ListItemProps) => {
  const classNames = getClassNames({
    className,
    conditionalClasses: { [`${listItemStyles.isLinks}`]: isLinks },
  })

  return (
    <li key={listKey} className={classNames} {...rest}>
      {children}
    </li>
  )
}

export default ListItem
