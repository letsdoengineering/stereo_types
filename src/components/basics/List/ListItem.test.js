import * as React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import ListItem from './ListItem'
import renderer from 'react-test-renderer'
import { axe } from 'jest-axe'

describe('ListItem', () => {
    it('renders li', async () => {
        render(<ListItem>a listitem</ListItem>)
        const list = screen.getByText('a listitem')

        await waitFor(() => expect(list.nodeName).toBe('LI'))
    })

    it('has not changed the JSX layout of the page from previous snapshot', () => {
        const tree = renderer.create(<ListItem />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('renders the page DOM that is accessible', async () => {
        const { container } = render(
            <ul>
                <ListItem>Some text</ListItem>
            </ul>
        )
        const results = await axe(container)

        expect(results).toHaveNoViolations()
    })
})
