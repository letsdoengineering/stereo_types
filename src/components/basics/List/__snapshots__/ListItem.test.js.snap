// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem has not changed the JSX layout of the page from previous snapshot 1`] = `
<li
  className=""
/>
`;
